lib/nodejs/esm-utils.js

```diff
@@ -92,9 +92,19 @@ const tryImportAndRequire = async (file, esmDecorator) => {
 const requireModule = async (file, esmDecorator) => {
   try {
     return require(file);
-  } catch (err) {
-    // Import if require fails.
-    return dealWithExports(await formattedImport(file, esmDecorator));
+  } catch (requireErr) {
+    // Fallback to import() if require() fails.
+    try {
+      return dealWithExports(await formattedImport(file, esmDecorator));
+    } catch (importErr) {
+      // If a --require module throws in a Node.js version that doesn't yet support .ts files,
```

Comment:
Maybe we can add tests for a specific range of Node versions? We should have an automated test that enters this branch of code for sure

Reply:
Ref also #5413 which also adds an if-else here and adds tests for the new branch